src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -1,162 +1,152 @@
-//        private JsonAddressBookIoExceptionThrowingStub(Path filePath) {
-//            super(filePath);
-//        }
-//
-//        @Override
-//        public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
-//            throw DUMMY_IO_EXCEPTION;
-//        }
-//    }
-//}
+package seedu.address.logic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static seedu.address.logic.commands.CommandTestUtil.MODULE_NAME_DESC_0;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalModules.MODULE_NAME_0;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import seedu.address.logic.commands.AddModuleCommand;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.HomeCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.ReadOnlyTeachingAssistantBuddy;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleName;
+import seedu.address.storage.JsonTeachingAssistantBuddyStorage;
+import seedu.address.storage.JsonUserPrefsStorage;
+import seedu.address.storage.StorageManager;
+
+public class LogicManagerTest {
+    private static final IOException DUMMY_IO_EXCEPTION = new IOException("dummy exception");
+
+    @TempDir
+    public Path temporaryFolder;
+
+    private Model model = new ModelManager();
+    private Logic logic;
+
+    @BeforeEach
+    public void setUp() {
+        JsonTeachingAssistantBuddyStorage assistantBuddyStorage =
+                new JsonTeachingAssistantBuddyStorage(temporaryFolder.resolve("assistantBuddy.json"));
+        JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(temporaryFolder.resolve("userPrefs.json"));
+        StorageManager storage = new StorageManager(assistantBuddyStorage, userPrefsStorage);
+        logic = new LogicManager(model, storage);
+    }
+
+    @Test
+    public void execute_invalidCommandFormat_throwsParseException() {
+        String invalidCommand = "uicfhmowqewca";
+        assertParseException(invalidCommand, MESSAGE_UNKNOWN_COMMAND);
+    }
+
+    @Test
+    public void execute_validCommand_success() throws Exception {
+        String listCommand = HomeCommand.COMMAND_WORD;
+        assertCommandSuccess(listCommand, HomeCommand.MESSAGE_SUCCESS, model);
+    }
+
+    @Test
+    public void execute_storageThrowsIoException_throwsCommandException() {
+        // Setup LogicManager with JsonAddressBookIoExceptionThrowingStub
```

Comment:
JsonTAB?

Reply:
Ah good catch! Ill fix this and then merge the PR!